code/datums/flock/flock.dm

```diff
@@ -409,14 +414,16 @@
 
 /datum/flock/proc/registerUnit(var/atom/movable/D)
 	if(isflock(D))
-		src.units |= D
+		if(!src.units[D.type])
+			src.units[D.type] = list()
```

Comment:
Maybe could just initialize flockbits too in New(), since not expecting anything else than flockdrones and flockbits atm, or was it to just to prevent something like retyping every unit from isflock() in New()? Probably fine with either, guess it just didn't seem necessary to do a check on if type exists when there's only a few types available.

Reply:
Not super needed, I only initialise the drone one because I directly reference it in code. If we need to reference bits also, we should do it then.